### PR TITLE
Merge with upstream

### PR DESCRIPTION
**Pull request by groyoh**
The `upstream` branch should now be used to be as close as possible to the upstream gem.

cc @runtastic/backend

### CODE REVIEW

File: test/collection_serializer_test.rb [hunk 1]
```diff
@@ -3,8 +3,11 @@
 module ActiveModel
   class Serializer
     class CollectionSerializerTest < ActiveSupport::TestCase
-      def setup
-        @comment = Comment.new
```
Comment (andreaseger) [inline test/collection_serializer_test.rb:7]: where was this class defined before?

Comment (groyoh) [inline test/collection_serializer_test.rb:7]: Which class?

File: test/adapter/json_api/has_many_test.rb [hunk 2]
```diff
@@ -4,6 +4,24 @@ module ActiveModelSerializers
   module Adapter
     class JsonApi
       class HasManyTest < ActiveSupport::TestCase
+        module Serializers
+          Post = Class.new(ActiveModel::Serializer) do
```
Comment (andreaseger) [inline test/adapter/json_api/has_many_test.rb]: I'd prefer defining these classes like this `class Post < ActiveModel::Serializer`

File: test/serializable_resource_test.rb [hunk 3]
```diff
@@ -2,6 +2,14 @@
 
 module ActiveModel
   class SerializableResourceTest < ActiveSupport::TestCase
+    module Serializers
+      Profile = Class.new(ActiveModel::Serializer) do
```
Comment (andreaseger) [inline test/serializable_resource_test.rb]: I'd prefer `class Profile < ActiveModel::Serializer`

File: test/serializers/associations_test.rb [hunk 4]
```diff
@@ -3,7 +3,21 @@
 module ActiveModel
   class Serializer
     class AssociationsTest < ActiveSupport::TestCase
-      def setup
+      module Serializers
+        Post = Class.new(ActiveModel::Serializer) do
```
Comment (andreaseger) [inline test/serializers/associations_test.rb]: I'd prefer `class Post < ActiveModel::Serializer`

File: test/collection_serializer_test.rb (hunk 1 above)
Comment (andreaseger) [inline test/collection_serializer_test.rb:7]: `Comment`

**Comment by andreaseger** [comment]
is the polymorphic support already in here? Haven't looked into the code outside this PR but at least the tests for it are not in here.

File: test/collection_serializer_test.rb (hunk 1 above)
Comment (groyoh) [inline test/collection_serializer_test.rb:7]: `Comment` is defined in `test/fixtures/poro.rb`

**Comment by groyoh** [comment]
Yes it's in there. It's basically the `namespace` option. Almost all the tests that were added test the behavior of this option. I will update the tests to properly demonstrate the polymorphism capability.

**Comment by groyoh** [comment]
Should be good now

**Comment by andreaseger** [comment]
:+1: It's fine by me. Unless someone else wants to review it too we can merge this PR.

**Comment by groyoh** [comment]
@andreaseger merge it whenever you feel like it or need it.

**Pull request merged**